op-challenger/game/scheduler/coordinator.go

```diff
@@ -33,24 +41,28 @@ type coordinator struct {
 	resultQueue <-chan job
 
 	logger       log.Logger
-	m            SchedulerMetricer
+	m            CoordinatorMetricer
 	createPlayer PlayerCreator
 	states       map[common.Address]*gameState
 	disk         DiskManager
+
+	// lastScheduledBlockNum is the highest block number that the coordinator has seen and scheduled jobs.
+	lastScheduledBlockNum uint64
 }
 
 // schedule takes the current list of games to attempt to progress, filters out games that have previous
 // progressions already in-flight and schedules jobs to progress on the outbound jobQueue.
 // To avoid deadlock, it may process results from the inbound resultQueue while adding jobs to the outbound jobQueue.
 // Returns an error if a game couldn't be scheduled because of an error. It will continue attempting to progress
 // all games even if an error occurs with one game.
-func (c *coordinator) schedule(ctx context.Context, games []types.GameMetadata) error {
+func (c *coordinator) schedule(ctx context.Context, games []types.GameMetadata, blockNumber uint64) error {
 	// First remove any game states we no longer require
 	for addr, state := range c.states {
 		if !state.inflight && !slices.ContainsFunc(games, func(candidate types.GameMetadata) bool {
 			return candidate.Proxy == addr
 		}) {
 			delete(c.states, addr)
+			continue
```

Comment:
nit: We can remove this continue again - at one point I was adding logic into this loop that needed to be skipped but that got removed so the continue is redundant again. Sorry, that's my mess getting left behind. :)